Made Several changes to as per #1

sudo-gyan opened the pull request:
Removed `build`, `dist` directories as they are generated from `setup.py` and committing them clutters code.

Worked on Fixing #1

TheSpeedX committed:
Made Several changes to as per #1

TheSpeedX committed:
added _generateJSON() for more flexibility

prashantsengar commented on Package/RedPy/redpy.py:
```diff
@@ -1,6 +1,10 @@
 import requests as _requests
 import os as _os
-import random
+from fake_useragent import UserAgent
+ua = UserAgent(
+    fallback=("Mozilla/5.0 (compatible; MSIE 10.0; Macintosh;"
+              " Intel Mac OS X 10_7_3; Trident/6.0"))
+
```
user agent hasn't been used anywhere

sudo-gyan replied:
Removed user agent

prashantsengar commented on Package/RedPy/redpy.py:
```diff
@@ -63,109 +49,42 @@ def _DownloadFiles(self, jsonfile, number_of_files):
-                videos.append(jsonfile['data']['children'][i]['data']['preview']['reddit_video_preview']['fallback_url'])
-            except Exception as e:
-                print(e)
-        return videos
+        flag = 1
```
This flag hasn't been defined anywhere in your code

sudo-gyan replied:
Yeah I added missing video download part  
And removed flag

prashantsengar reviewed (commented):
@sudo-gyan cool work! I have added some comments, please fix those and we are ready to merge!

TheSpeedX committed:
added previous video download method and removed fake ua

prashantsengar commented on Package/RedPy/redpy.py:
```diff
@@ -26,146 +27,121 @@ def download(self, subreddit, number=5, sort_option=None,
 
-        self.url = 'https://www.reddit.com/r/' + subreddit + '/'+  sort_option  +  '.json'
-        print(self.url)
+        json_data = self._generateJSON(subreddit, sort_option)
```
in `_generateJSON` function, if there is an error in the request, you return `None`. You are not handling this `None` anywhere. 
This will cause error since `json_data ` will be None and you pass this to the next functions.

We are actually handling this with the `try-except` block in the getMedia functions but there should be a better way of handling this.

Maybe raise an exception there when the request is 200. Or check if `json_data` is None and return/raise and ask the user to use another user-agent.

Perhaps raising and exception would be a better option

sudo-gyan replied:
Modified to raise exception in such errors

prashantsengar commented on Package/RedPy/redpy.py:
```diff
@@ -26,146 +27,121 @@ def download(self, subreddit, number=5, sort_option=None,
 
-        self.url = 'https://www.reddit.com/r/' + subreddit + '/'+  sort_option  +  '.json'
-        print(self.url)
+        json_data = self._generateJSON(subreddit, sort_option)
```
Why are we calling the `_generateJSON` function in both the download media functions? Why not call it once in the download function? Is there any reason you chose this?

prashantsengar replied:
I think you are doing this because the function is public.

Maybe you can have a default parameter in the function and check its value. If it's empty, then you can call the `generateJSON` function in the separate functions

sudo-gyan replied:
Added `json_data` to member variable so as to call once.
The `_getImages()` and `_getVideos()` still take `json_data` as parameter so as others can parse their own `json_data` if required.
And its used in RedIns as well

sudo-gyan committed:
error handling in json and calling it once

prashantsengar committed:
Fixed small issues

- in `download`, `sort_option`'s default value is None while later it's default value is change to empty string. Hence, it was not being checked if it is `None`. I have added a check in every `download` function and changed its default value to `None` everywhere for better understandabilty. (sort_option=None is easier to understand than keeping it empty string)
- Added context file handlers as recommended by PEP
- Changed `user` to `user_agent` in `init`, easier to understand by the end user

prashantsengar commented:
I have added some minor changes @sudo-gyan , make sure to pull the changes
